iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java

```diff
@@ -587,6 +590,80 @@ CalculatedBusTopology getCalculatedBusTopology() {
     }
 
     private final NodeBreakerViewExt nodeBreakerView = new NodeBreakerViewExt() {
+
+        private final Map<Integer, TDoubleArrayList> fictitiousP0ByNode = new HashMap<>();
+        private final Map<Integer, TDoubleArrayList> fictitiousQ0ByNode = new HashMap<>();
+
+        @Override
+        public double getFictitiousP0(int node) {
+            if (fictitiousP0ByNode.containsKey(node)) {
+                return fictitiousP0ByNode.get(node).get(getNetwork().getVariantIndex());
+            }
+            return 0.0;
+        }
+
+        @Override
+        public NodeBreakerView setFictitiousP0(int node, double p0) {
+            if (Double.isNaN(p0)) {
+                throw new ValidationException(NodeBreakerVoltageLevel.this, "undefined value cannot be set as fictitious p0");
+            }
+            TDoubleArrayList p0ByVariant = fictitiousP0ByNode.computeIfAbsent(node, n -> {
```

Comment:
Note that it introduces a "race condition" between variants, if called in multithread:
if 2 threads try to set p0, we might (although unlikely) create the list in the 2 threads, and end up with only once of them, so one of the values will be ignored.

Note sure of the best solution:
- should we simply ignore it ?
- should we synchronize the map ?
- should we invert the implementation and use one list of doubles by variant ?
- should we follow implementation for V/angle which puts it in node terminal?
- ...

Reply:
Following your review I made the maps synchronized. What do you think?

Reply:
Ok!